File: app/src/main/java/com/nfcalarmclock/permission/postnotifications/NacPostNotificationsPermissionRequestDialog.java
package com.nfcalarmclock.permission.postnotifications;

import android.Manifest;
import android.os.Build;
import androidx.annotation.RequiresApi;
import com.nfcalarmclock.R;
import com.nfcalarmclock.permission.NacPermissionRequestDialog;
import com.nfcalarmclock.shared.NacSharedPreferences;

/**
 * Dialog to request to allow posting notifications.
 */
@RequiresApi(api=Build.VERSION_CODES.TIRAMISU)
public class NacPostNotificationsPermissionRequestDialog
	extends NacPermissionRequestDialog
{

	/**
	 * Tag for the class.
	 */
	public static final String TAG = "NacPostNotificationsPermissionDialog";

	/**
	 * The actions to execute when the permission request is accepted.
	 */
	protected void doPermissionRequestAccepted()
	{
		NacSharedPreferences shared = this.getSharedPreferences();

		// Set the flag that the permission was requested
		shared.editWasPostNotificationsPermissionRequested(true);

		// Call the accepeted listeners
		super.doPermissionRequestAccepted();
	}

	/**
	 * The actions to execute when the permission request is canceled.
	 */
	protected void doPermissionRequestCanceled()
	{
		NacSharedPreferences shared = this.getSharedPreferences();

		// Set the flag that the permission was requested
		shared.editWasPostNotificationsPermissionRequested(true);

		// Call the canceled listeners
		super.doPermissionRequestCanceled();
	}

	/**
	 * Get the name of the permission.
	 *
	 * @return The name of the permission.
	 */
	public String getPermission()
	{
		return Manifest.permission.POST_NOTIFICATIONS;
	}

	/**
	 * Get the ID of the layout.
	 *
	 * @return The ID of the layout.
	 */
	public int getLayoutId()
	{
		return R.layout.dlg_request_post_notifications_permission;
	}

	/**
	 * Get the ID of the title string.
	 *
	 * @return The ID of the title string.
	 */
	public int getTitleId()
	{
		return R.string.title_request_permission_post_notifications;
	}

}

File: app/src/main/kotlin/com/nfcalarmclock/system/permission/postnotifications/NacPostNotificationsPermissionRequestDialog.kt
package com.nfcalarmclock.system.permission.postnotifications

import android.os.Build
import androidx.annotation.RequiresApi
import com.nfcalarmclock.R
import com.nfcalarmclock.system.permission.NacPermissionRequestDialog

/**
 * Dialog to request to allow posting notifications.
 */
@RequiresApi(api = Build.VERSION_CODES.TIRAMISU)
class NacPostNotificationsPermissionRequestDialog
	: NacPermissionRequestDialog()
{

	/**
	 * The actions to execute when the permission request is accepted.
	 */
	override fun doPermissionRequestAccepted()
	{
		// Set the flag that the permission was requested
		sharedPreferences!!.wasPostNotificationsPermissionRequested = true

		// Call the accepeted listeners
		super.doPermissionRequestAccepted()
	}

	/**
	 * The actions to execute when the permission request is canceled.
	 */
	override fun doPermissionRequestCanceled()
	{
		// Set the flag that the permission was requested
		sharedPreferences!!.wasPostNotificationsPermissionRequested = true

		// Call the canceled listeners
		super.doPermissionRequestCanceled()
	}

	/**
	 * The name of the permission.
	 */
	override val permission: String
		get() = NacPostNotificationsPermission.permissionName

	/**
	 * The ID of the layout.
	 */
	override val layoutId: Int
		get() = R.layout.dlg_request_post_notifications_permission

	/**
	 * The ID of the title string.
	 */
	override val titleId: Int
		get() = R.string.title_request_permission_post_notifications

	companion object
	{

		/**
		 * Tag for the class.
		 */
		const val TAG = "NacPostNotificationsPermissionDialog"

	}

}